src/main/java/uk/gov/hmcts/darts/authorisation/config/AuthorisationAspect.java

```diff
@@ -0,0 +1,262 @@
+            case CASE_ID -> checkAuthorisationByCaseId(request, roles);
+            case HEARING_ID -> checkAuthorisationByHearingId(request, roles);
+            case MEDIA_REQUEST_ID -> checkAuthorisationByMediaRequestId(request, roles);
+            case MEDIA_ID -> checkAuthorisationByMediaId(request, roles);
+            case TRANSCRIPTION_ID -> checkAuthorisationByTranscriptionId(request, roles);
+            default -> throw new IllegalStateException(String.format(
+                "The Authorisation annotation contextId is not known: %s",
+                contextId
+            ));
+        }
+    }
+
+    private void checkAuthorisationByCaseId(HttpServletRequest request, Set<SecurityRoleEnum> roles) {
+        Optional<String> caseIdParamOptional = Optional.empty();
+
+        Matcher matcher = CASES_ID_PATH_PATTERN.matcher(request.getRequestURI());
+        if (matcher.find()) {
+            caseIdParamOptional = Optional.ofNullable(matcher.group(0));
+            checkAuthorisationByCaseId(caseIdParamOptional, roles);
+        }
+
+        if (caseIdParamOptional.isEmpty()) {
+            caseIdParamOptional = Optional.ofNullable(request.getParameter(CASE_ID_PARAM));
+            checkAuthorisationByCaseId(caseIdParamOptional, roles);
+        }
+
+        if (caseIdParamOptional.isEmpty()) {
+            caseIdParamOptional = Optional.ofNullable(request.getHeader(CASE_ID_PARAM));
+            checkAuthorisationByCaseId(caseIdParamOptional, roles);
+        }
+    }
+
+    private void checkAuthorisationByCaseId(Optional<String> caseIdParamOptional, Set<SecurityRoleEnum> roles) {
+        if (caseIdParamOptional.isPresent()) {
+            try {
+                Integer caseId = Integer.valueOf(caseIdParamOptional.get());
+                authorisation.authoriseByCaseId(caseId, roles);
+            } catch (NumberFormatException e) {
+                log.error("Unable to parse case_id for checkAuthorisation", e);
+                throw new DartsApiException(BAD_REQUEST_CASE_ID);
+            }
+        }
+    }
+
+    private void checkAuthorisationByHearingId(HttpServletRequest request, Set<SecurityRoleEnum> roles) {
+        Optional<String> hearingIdParamOptional = Optional.empty();
+
+        Matcher matcher = HEARINGS_ID_PATH_PATTERN.matcher(request.getRequestURI());
+        if (matcher.find()) {
+            hearingIdParamOptional = Optional.ofNullable(matcher.group(0));
+            checkAuthorisationByHearingId(hearingIdParamOptional, roles);
+        }
+
+        if (hearingIdParamOptional.isEmpty()) {
+            hearingIdParamOptional = Optional.ofNullable(request.getParameter(HEARING_ID_PARAM));
+            checkAuthorisationByHearingId(hearingIdParamOptional, roles);
+        }
+
+        if (hearingIdParamOptional.isEmpty()) {
+            hearingIdParamOptional = Optional.ofNullable(request.getHeader(HEARING_ID_PARAM));
+            checkAuthorisationByHearingId(hearingIdParamOptional, roles);
+        }
```

Comment:
I guess the same comment applies about if your authentication is against hearing id, media request id etc and its null, should it fail authentication. Not sure how you handle the requests that need to create the hearing, case, media etc.

Reply:
Done in commit https://github.com/hmcts/darts-api/commit/5af180b88f1eb52400920bfeba84714892a24238